### PR TITLE
Conan Fixups Part III

### PR DESCRIPTION
**Pull request by johannes-wolf**
This PR:
- Guards all `FetchContent` calls, so that they do not get called if `SIMFIL_CONAN` is set
- Removes all the `FIND_PACKAGE_ARGS` arguments
- Downgrades `fmt` to `10.0.0` because of compat. with `spdlog` in mapget – I guess it would be best to not have `fmt` in the `simfil` public-headers, though
- Add guards for each target (TODO: check case of the target names!)

### CODE REVIEW

**Comment by josephbirkner** [comment]
In the mapget relations branch, the fmt compatibility problem is solved by setting `SPDLOG_FMT_EXTERNAL` before fetchcontenting spdlog. So we don't need to downgrade fmt imo 🤔 

```
set(SPDLOG_FMT_EXTERNAL TRUE CACHE STRING "Ensure that spdlog uses fmt from conan.")
```

File: deps.cmake [hunk 1]
```diff
@@ -7,42 +7,48 @@ if (SIMFIL_CONAN)
   if (SIMFIL_WITH_MODEL_JSON)
     find_package(nlohmann_json REQUIRED)
   endif()
```
Comment (josephbirkner) [inline deps.cmake:9]: Should we also find_package(catch2) if SIMFIL_WITH_TESTS?

Comment (johannes-wolf) [reply deps.cmake:9]: Conans Catch2 is broken.

**Comment by johannes-wolf** [comment]
> In the mapget relations branch, the fmt compatibility problem is solved by setting `SPDLOG_FMT_EXTERNAL` before fetchcontenting spdlog. So we don't need to downgrade fmt imo 🤔
> 
> ```
> set(SPDLOG_FMT_EXTERNAL TRUE CACHE STRING "Ensure that spdlog uses fmt from conan.")
> ```

But if both, fmt and spdlog are from conan, setting this variable does nothing.

**Comment by josephbirkner** [comment]
> > In the mapget relations branch, the fmt compatibility problem is solved by setting `SPDLOG_FMT_EXTERNAL` before fetchcontenting spdlog. So we don't need to downgrade fmt imo 🤔
> > ```
> > set(SPDLOG_FMT_EXTERNAL TRUE CACHE STRING "Ensure that spdlog uses fmt from conan.")
> > ```
> 
> But if both, fmt and spdlog are from conan, setting this variable does nothing.

Makes sense!

**Review by josephbirkner — APPROVED** [review]
LGTM

File: deps.cmake [hunk 2]
```diff
@@ -7,42 +7,48 @@ if (SIMFIL_CONAN)
...
-FetchContent_Declare(sfl
-  GIT_REPOSITORY "https://github.com/slavenf/sfl-library.git"
-  GIT_TAG        "master"
-  GIT_SHALLOW    ON
-  FIND_PACKAGE_ARGS)
+else()
+  if (NOT TARGET sfl::sfl)
+    FetchContent_Declare(sfl
+      GIT_REPOSITORY "https://github.com/slavenf/sfl-library.git"
+      GIT_TAG        "master"
```
Comment (johannes-wolf) [inline deps.cmake:14]: TODO: Pin a release

**Comment by johannes-wolf** [comment]
Will merge this after applying some of the changes from the conan-io simfil recipe, which got merged! :tada:

**Comment by josephbirkner** [comment]
I have a question, which occured to me earlier. Or, more like a series of thoughts, haha. Wouldn't it be better, to use a conanfile.txt (+cmake-conan) for local development instead of FetchContent? And then consume the dependencies listed in conanfile.txt in conanfile.py? Or, to put it differently, when do you think we can get rid of the FetchContent-based dependency fetching? Is it only needed while we build up the tree of conan packages?

**Comment by johannes-wolf** [comment]
With a `conanfile.txt` you canot use simfil as `editable` package. The `FetchContent` should get removed as soon as possible. It exists to not break packages which `FetchContent(simfil ... GIT_TAG "master")`. It has nothing to do with local dev.

I am against parsing the conanfile.txt insides the conanfile.py.

**Comment by josephbirkner** [comment]
> With a `conanfile.txt` you canot use simfil as `editable` package. The _`FetchContent` should get removed as soon as possible_! It exists to not break packages which `FetchContent(simfil ... GIT_TAG "master")`. It has nothing to do with local dev.
> 
> 
> 
> I am against parsing the conanfile.txt insides the conanfile.py.
> 
> 

Thanks for the explanation 👌

**Pull request merged**